controller/volume_controller_test.go

```diff
@@ -119,15 +119,38 @@ func (s *TestSuite) TestVolumeLifeCycle(c *C) {
 		tc.nodes[i].Spec.AllowScheduling = false
 	}
 	tc.copyCurrentToExpect()
-	// replicas and engine object would still be created
+	// replica object would not be created
 	tc.replicas = nil
+	// engine object would still be created
 	tc.engines = nil
 	for _, r := range tc.expectReplicas {
 		r.Spec.NodeID = ""
 		r.Spec.DiskID = ""
 		r.Spec.DiskPath = ""
 		r.Spec.DataDirectoryName = ""
 	}
+	tc.expectVolume.Status.State = longhorn.VolumeStateCreating
+	tc.expectVolume.Status.CurrentImage = tc.volume.Spec.Image
+	tc.expectVolume.Status.Robustness = longhorn.VolumeRobustnessFaulted
+	testCases["volume create - replica creation failure"] = tc
+
+	// unable to create volume because no node to schedule
```

Comment:
Can you help me to understand this test case? If I understand correctly, the one above (`volume create - replica creation failure`) is now expected. (We will try not to create replicas that cannot be scheduled.) In this test case, the replicas are already created but cannot be scheduled (as was the old behavior). Is this a state we expect to be in? Or is it just to cover a corner case that probably won't be triggered?

Reply:
This is likely a corner case. In my opinion, the unit test will give us a better understanding of the behavior.